dotnet/src/SemanticKernel.Abstractions/PromptTemplate/PromptTemplateConfig.cs

```diff
@@ -136,7 +136,7 @@ internal IReadOnlyList<KernelParameterMetadata> GetKernelParametersMetadata()
                 Description = p.Description,
                 DefaultValue = p.Default,
                 IsRequired = p.IsRequired,
-                ParameterType = !string.IsNullOrWhiteSpace(p.JsonSchema) ? null : typeof(string),
+                ParameterType = !string.IsNullOrWhiteSpace(p.JsonSchema) ? null : p.Default?.GetType(),
```

Comment:
If there's a null default, should it default to string? Otherwise, what happens for inferred inputs?

Reply:
To be honest, I'm not sure that it should default to a string. Currently, both the `KernelPromptTemplate` and `HandlebarsPromptTemplate` classes support string and non-string data types, such as primitives, arrays, and objects. Neither class infers the types of prompt variables - https://github.com/microsoft/semantic-kernel/issues/4350. Defaulting to a string could be misleading since it may not match the actual type of the prompt variables, the type of which is impossible to infer—see the 'idea' variable in the issue mentioned above. As such, null would be a more appropriate indicator of an unknown type. However, there is no harm in defaulting to a string since there's logic in place to convert string arguments to their respective function parameter types, which would be triggered if string arguments were provided for non-string function parameters. For now, I will likely add this fallback to a string to be on the safe side, and we might revise the logic later.

Reply:
Ok. We very recently added the string default fallback, but we can revisit if there's a better answer. For that particular motivating case, I believe we also added a string fallback to openai, so it shouldn't actually matter for that.